contrib/platform/src/com/sun/jna/platform/win32/Advapi32Util.java

```diff
@@ -1540,16 +1532,9 @@ public static void registrySetStringValue(HKEY hKey, String name,
         if (value == null) {
             value = "";
         }
-        Memory data;
-        if (W32APITypeMapper.DEFAULT == W32APITypeMapper.UNICODE) {
-            data = new Memory((value.length() + 1) * Native.WCHAR_SIZE);
-            data.setWideString(0, value);
-        } else {
-            data = new Memory((value.length() + 1));
-            data.setString(0, value);
-        }
-        int rc = Advapi32.INSTANCE.RegSetValueEx(hKey, name, 0, WinNT.REG_SZ,
-            data, (int) data.size());
+        NativeWin32String data = new NativeWin32String(value);
```

Comment:
You're missing the extra null character in this allocation. Which you probably don't need to change as the previous version properly handled unicode.